src/dawn/engine_dawn.cc

```diff
@@ -712,29 +757,29 @@ Result DawnPipelineHelper::CreateRenderPipelineDescriptor(
   renderPipelineDescriptor.fragmentStage = std::move(&fragmentStage);
 
   // Set defaults for the input state descriptors.
-  tempInputState.indexFormat = ::dawn::IndexFormat::Uint32;
-  // Fill the default values for vertexInput.
-  vertexInput.inputSlot = 0;
-  vertexInput.stride = 0;
-  vertexInput.stepMode = ::dawn::InputStepMode::Vertex;
-  for (uint32_t i = 0; i < kMaxVertexInputs; ++i) {
-    tempInputs[i] = vertexInput;
+  vertexInputDescriptor = new ComboVertexInputDescriptor();
```

Comment:
It seems unusual to have an allocation of this object, and then hand ownership over to the Dawn API.  I guess this leaks.  I would expect this to be allocated on the stack and you pass in the address to the .vertexInput member of renderPipelineDiscriptor ??